evm/finality-block.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    providers: { JsonRpcProvider },
+} = ethers;
+const { Command, Option } = require('commander');
+const { printInfo, mainProcessor } = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+
+async function processCommand(config, chain, options) {
+    let { rpc, waitTime, update } = options;
+
+    if (!rpc) {
+        rpc = chain.rpc;
+    }
+
+    let max = 0;
+    let sum = 0;
+    const DELAY = 10000;
+
+    const provider = new JsonRpcProvider(rpc);
+
+    try {
+        await provider.getBlock('finalized');
+    } catch {
+        if (update && waitTime) {
+            chain.finality = parseInt(waitTime);
+        }
+
+        printInfo('Finalized tag not supported by rpc');
+        return;
+    }
+
+    for (let i = 0; i < 10; i++) {
+        const latestBlockPromise = provider.getBlock('latest');
+        const finalizedBlockPromise = provider.getBlock('finalized');
+
+        const latestBlock = await latestBlockPromise;
+        const finalizedBlock = await finalizedBlockPromise;
```

Comment:
```suggestion
        const latestBlock = await provider.getBlock('latest');
        const finalizedBlock = await provider.getBlock('finalized');
```
And remove lines 36-37?

Reply:
i suppose this was done for concurrency, so that both requests are made simultaneously without waiting on one, to give a better measure

Reply:
yes, as this would decrease time between the request for blocks and the difference would be more precise.